handler.go

```diff
@@ -10,11 +10,22 @@ import (
 )
 
 // Logger defines how HTTP requests are logged, e.g. to the console, or in JSON format (see JSONLogger).
-type Logger func(method string, url *url.URL, status int, len int64, d time.Duration)
+type Logger func(r *http.Request, status int, len int64, d time.Duration)
 
-// JSONLogger uses logrus to log the HTTP request in JSON format to os.Stderr.
-func JSONLogger(method string, url *url.URL, status int, len int64, d time.Duration) {
-	os.Stderr.WriteString(JSONLogMessage(time.Now, method, url, status, len, d))
+// JSONLogger logs the HTTP request in JSON format to os.Stderr.
+func JSONLogger(r *http.Request, status int, len int64, d time.Duration) {
+	os.Stderr.WriteString(JSONLogMessage(time.Now, r.Method, r.URL, status, len, d, nil))
+}
+
+// NewJSONLoggerWithHeaders returns a logger that logs the given headers of an HTTP request.
+func NewJSONLoggerWithHeaders(h []string) Logger {
```

Comment:
That will be more user friendly like ` func NewJSONLoggerWithHeaders(h ...string) Logger { `